### PR TITLE
Add docs on repository reliability and backups

### PR DESCRIPTION
**Pull request by DaveCTurner**
This commit spells out how important repository reliability is to
searchable snapshots, and also documents a procedure for taking a backup
of a snapshot repository.

Relates #54944

### CODE REVIEW

File: docs/reference/snapshot-restore/register-repository.asciidoc [hunk 1]
```diff
@@ -294,3 +294,27 @@ subsequently cleaned up.
 Please note that most of the cleanup operations executed by this endpoint are automatically executed when deleting any snapshot from a
 repository. If you regularly delete snapshots, you will in most cases not get any or only minor space savings from using this functionality
 and should lower your frequency of invoking it accordingly.
+
+[discrete]
+[[snapshots-repository-backup]]
+=== Repository backup
```
Comment (DaveCTurner) [inline docs/reference/snapshot-restore/register-repository.asciidoc:300]: I'm in two minds about this section. We don't have any tests to support these claims today AFAIK, and it seems like they might be pretty tricky to write. How comfortable are we with writing this in the docs without such tests?

Comment (DaveCTurner) [reply docs/reference/snapshot-restore/register-repository.asciidoc:300]: I have an idea for a way to write tests for this but have encountered some unrelated issues along the way.

**Review by ywelsch — APPROVED** [review]
LGTM

File: docs/reference/snapshot-restore/register-repository.asciidoc (hunk 1 above)
Comment (ywelsch) [reply docs/reference/snapshot-restore/register-repository.asciidoc:300]: I'm ok with this section. I'm not sure what you would like to test here. Isn't it just about having a repo mounted, then unmount it, and then remount it and see if everything works?

Comment (DaveCTurner) [reply docs/reference/snapshot-restore/register-repository.asciidoc:300]: No I'd like to take a copy of the repo either (a) atomically, while snapshots/deletes/... are ongoing or (b) while it's readonly and verify that the copy is usable as a repository too. You can only unregister a repository while there's no ongoing activity which is a much weaker claim.

**Pull request merged**